internal/btf/types.go

```diff
@@ -528,7 +535,20 @@ func inflateRawTypes(rawTypes []rawType, rawStrings stringTable) (namedTypes map
 			typ = fn
 
 		case kindFuncProto:
-			fp := &FuncProto{id, nil}
+			rawparams := raw.data.([]btfParam)
+			params := make([]FuncParam, 0, len(rawparams))
+			for _, param := range rawparams {
+				// OK for func params to not have a name
+				name, _ := rawStrings.LookupName(param.NameOff)
```

Comment:
From `Documentation/bpf/btf.rst`:
```rst
If a BTF_KIND_FUNC_PROTO type is referred by a BTF_KIND_FUNC type, then
``btf_param.name_off`` must point to a valid C identifier except for the
possible last argument representing the variable argument. The btf_param.type
refers to parameter type.

If the function has variable arguments, the last parameter is encoded with
``name_off = 0`` and ``type = 0``.
```

We should only try to resolve a name if NameOff != 0. From the description it sounds a bit like the rule doesn't apply if FUNC_PROTO is not referred to by a FUNC, can you tell me more about this?

Reply:
From the tests, it looks like attempting to resolve `0` is actually OK, and results in `""`. I can reinstate the error handling. https://github.com/cilium/ebpf/blob/4048cd641690f2f82d57b130467a3fdfe4967a06/internal/btf/strings_test.go#L25

Reply:
Updated.

Reply:
Yes, offset 0 is guaranteed to be the empty string by the spec.